proto/structured.proto

```diff
@@ -0,0 +1,95 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Tamir Duberstein (tamird@gmail.com)
+
+syntax = "proto2";
+package cockroach.proto;
+option go_package = "proto";
+
+import "gogoproto/gogo.proto";
+import "cockroach/proto/api.proto";
+import "cockroach/proto/errors.proto";
+
+option (gogoproto.sizer_all) = true;
+option (gogoproto.marshaler_all) = true;
+option (gogoproto.unmarshaler_all) = true;
+
+message Column {
+  enum ColumnType {
+    // option (gogoproto.goproto_enum_prefix) = false;
+    BYTES = 0;
+  }
+
+  optional string name = 1 [(gogoproto.nullable) = false];
+  optional ColumnType type = 2 [(gogoproto.nullable) = false];
+}
+
+message Index {
+  optional string name = 1 [(gogoproto.nullable) = false];
+  optional bool unique = 2 [(gogoproto.nullable) = false];
+}
+
+message TableSchema {
+  message IndexByName {
```

Comment:
Fixed (by using IndexByName in the list of indexes). I'd like to stick with storing ids for now since it's more consistent, and we can hide all these details with accessor methods.